modules/lti-service-remote/src/main/java/org/opencastproject/lti/endpoint/LtiServiceRemoteImpl.java

```diff
@@ -0,0 +1,178 @@
+import org.opencastproject.security.api.UnauthorizedException;
+import org.opencastproject.serviceregistry.api.RemoteBase;
+import org.opencastproject.util.NotFoundException;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.entity.mime.content.InputStreamBody;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import javax.ws.rs.core.Response;
+
+/**
+ * The service calling the LTI REST endpoint (for multi-node setups with LTI)
+ */
+public class LtiServiceRemoteImpl extends RemoteBase implements LtiService {
+  private static final Gson gson = new Gson();
+
+  public LtiServiceRemoteImpl() {
+    super(LtiService.JOB_TYPE);
+  }
+
+  private HttpResponse safeGetResponse(final HttpRequestBase r) {
+    final HttpResponse response = getResponse(r);
+    if (response == null) {
+      throw new RuntimeException("No response from service");
+    }
+    return response;
+  }
+
+  @Override
+  public List<LtiJob> listJobs(final String seriesId) {
+    HttpResponse response = null;
+    try {
+      response = safeGetResponse(new HttpGet("/jobs?seriesId=" + seriesId));
+      return gson.fromJson(
+              new InputStreamReader(response.getEntity().getContent(), StandardCharsets.UTF_8),
+              new TypeToken<List<LtiJob>>() {
+              }.getType());
+    } catch (IOException e) {
+      throw new RuntimeException("failed retrieving jobs", e);
+    } finally {
+      closeConnection(response);
+    }
+  }
+
+  @Override
+  public void upsertEvent(
+          final LtiFileUpload file,
+          final String captions,
+          final String eventId,
+          final String seriesId,
+          final String metadataJson) {
+    final MultipartEntityBuilder entity = MultipartEntityBuilder.create();
+    entity.addTextBody("isPartOf", seriesId);
+    entity.addTextBody("metadata", metadataJson);
+    if (eventId != null) {
+      entity.addTextBody("eventId", eventId);
+    }
+    if (captions != null) {
+      entity.addTextBody("captions", captions);
+    }
+    if (file != null) {
+      entity.addPart(file.getSourceName(), new InputStreamBody(file.getStream(), file.getSourceName()));
+    }
+    final HttpPost post = new HttpPost("/");
+    post.setEntity(entity.build());
+    closeConnection(safeGetResponse(post));
+  }
+
+  @Override
+  public void copyEventToSeries(final String eventId, final String seriesId) {
+    final HttpPost get = new HttpPost("/" + eventId + "/copy?seriesId=" + seriesId);
+    final HttpResponse response = getResponse(get);
```

Comment:
This isn't using `safeGetResponse`.  Also, doesn't `getReponse` check for a 200, whereas the copy endpoint returns a 204?